controllers/logging/secret_controller.go

```diff
@@ -0,0 +1,76 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/ViaQ/logerr/log"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	// "sigs.k8s.io/controller-runtime/types"
+
+	loggingv1 "github.com/openshift/elasticsearch-operator/apis/logging/v1"
+)
+
+// SecretReconciler reconciles a Secret object
+type SecretReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+func (r *SecretReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	_ = context.Background()
+
+	cluster := &loggingv1.Elasticsearch{}
+	esName := types.NamespacedName{
+		// TODO: how to avoid hard coding the cluster NamespaceName?
+		Namespace: "openshift-logging",
+		Name:      "elasticsearch",
```

Comment:
we don't need to hard code this, the name and namespace should match the secret that we are reconciling on